src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -102,23 +191,41 @@ public override int GetHashCode()
             return hash;
```

Comment:
This hashcode allocates. This is really unfortunate. We should make sure this isn't showing up in our profiles.